src/lib/y2network/udev_rule_part.rb

```diff
@@ -0,0 +1,81 @@
+# Copyright (c) [2019] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+module Y2Network
+  # Simple class to represent a key-value pair in a udev rule
+  #
+  # This class does not check whether operators or keys/values are valid or not. We can implement
+  # that logic later if required.
+  class UdevRulePart
+    # Regular expression to match a udev rule part
+    PART_REGEXP = Regexp.new("\\A(?<key>[A-Za-z\{\}]+)(?<operator>[^\"]+)\"(?<value>.+)\"\\Z")
+
+    class << self
+      # Returns a rule part from a string
+      #
+      # @example Simple case
+      #   part = UdevRulePart.from_string('ACTION=="add"')
```

Comment:
is it mandatory to have key without quotes and value with quotes? according to `man udev` Or is it defined by our agent for reading it?

Reply:
I am following our agent convention but I also checked some examples here and there and it looks like the right syntax. Even rules under `/etc/udev/rules.d` follow this syntax.